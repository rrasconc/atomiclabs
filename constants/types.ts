import type { FontAwesome } from '@expo/vector-icons';
import type { NativeStackScreenProps } from '@react-navigation/native-stack';
import type { FormikErrors, FormikHandlers } from 'formik';
import type { ReactNode } from 'react';
import type {
  ImageSourcePropType,
  TextInputProps as DefaultTextInputProps,
  TouchableOpacityProps,
  ScrollViewProps,
  ColorValue,
  LayoutChangeEvent,
} from 'react-native';

export interface ServiceGroup {
  title: string;
  services: string[];
  image: ImageSourcePropType;
}

export interface ServiceGroupCardProps extends ServiceGroup {
  width: number;
}

export interface TeamMember {
  id: number;
  avatar: string;
  name: string;
  position: string;
}

export type TeamMemberCardProps = Pick<TeamMember, 'avatar' | 'name' | 'position'>;

export interface TeamMembersProps {
  data: TeamMember[];
  loading: boolean;
  error: any;
}

type FontawesomeIcon = keyof typeof FontAwesome.glyphMap;

export interface SocialMediaButtonProps {
  icon: FontawesomeIcon;
  url: string;
}

export interface ScrollDownButtonProps extends TouchableOpacityProps {
  text: string;
}

type RootStackParamList = {
  Home: undefined;
  SignUp: undefined;
};

export type ScreenProps = NativeStackScreenProps<RootStackParamList>;

export type StepStatus = 'pending' | 'inProgress' | 'completed';

export interface StepDotProps {
  status: StepStatus;
  size?: 'small' | 'large';
  number: number;
  color?: ColorValue;
}

export interface TextInputProps extends DefaultTextInputProps {
  label: string;
  error: string;
}

export interface ButtonProps extends TouchableOpacityProps {
  text: string;
  loading?: boolean;
}

export interface StepperProps {
  children: ReactNode;
  progress: number;
}

export interface SignUpFormValues {
  name: string;
  lastname: string;
  phoneNumber: string;
}

export interface GenericForm {
  loading?: boolean;
}

export interface NameFormProps
  extends GenericForm,
    Pick<FormikHandlers, 'handleSubmit' | 'handleChange'> {
  errors: FormikErrors<SignUpFormValues>;
  values: SignUpFormValues;
}

export interface PhoneFormProps
  extends GenericForm,
    Pick<FormikHandlers, 'handleSubmit' | 'handleChange'> {
  errors: FormikErrors<SignUpFormValues>;
  values: SignUpFormValues;
}

export interface GallaxyScrollViewProps extends ScrollViewProps {
  renderBgImages: () => ReactNode;
}

export interface ServicesProps {
  data: ServiceGroup[];
  onLayout: (e: LayoutChangeEvent) => void;
}

export interface ServiceCarouselProps {
  data: ServiceGroup[];
}
