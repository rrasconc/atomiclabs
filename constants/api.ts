import axios from 'axios';

export const api = axios.create({
  baseURL: 'https://atomic-test-api.onrender.com',
});
