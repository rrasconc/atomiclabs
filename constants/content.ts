import type { ServiceGroup } from './types';

const conquistaImg = require('../assets/conquista.png');
const exploraImg = require('../assets/explora.png');
const imaginaImg = require('../assets/imagina.png');

export const SERVICE_GROUPS: ServiceGroup[] = [
  {
    title: 'IMAGINA',
    services: [
      'Estrategia Digital',
      'Big Data & Analysis',
      'Consultoría Tecnológica',
      'Reducción de costos TI',
    ],
    image: imaginaImg,
  },
  {
    title: 'EXPLORA',
    services: ['Innovación y creación tecnológica', 'UI/UX', 'Innovación'],
    image: exploraImg,
  },
  {
    title: 'CONQUISTA',
    services: [
      'Desarrollo tecnológico y creación tecnológica',
      'Ciberseguridad',
      'Servicios de la Nube',
    ],
    image: conquistaImg,
  },
];
