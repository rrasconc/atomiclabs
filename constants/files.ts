const ASTRO1_IMG = require('../assets/astro1.png');
const ASTRO2_IMG = require('../assets/astro2.png');
const ASTRO3_IMG = require('../assets/astro3.png');
const BUBBLES_IMG = require('../assets/bubbles.png');
const GALLAXY_IMG = require('../assets/gallaxy.png');
const GALLAXY_2_IMG = require('../assets/gallaxy2.png');
const GALLAXY_3_IMG = require('../assets/gallaxy3.png');
const LOGO_IMG = require('../assets/logo.png');
const PEOPLE_IMG = require('../assets/people.png');
const SMALL_BUBBLES_IMG = require('../assets/small_bubbles.png');
const SMALL_GALLAXY_IMG = require('../assets/small_gallaxy.png');

export const IMAGES = {
  astro1: ASTRO1_IMG,
  astro2: ASTRO2_IMG,
  astro3: ASTRO3_IMG,
  bubbles: BUBBLES_IMG,
  gallaxy: GALLAXY_IMG,
  gallaxy2: GALLAXY_2_IMG,
  gallaxy3: GALLAXY_3_IMG,
  smallBubbles: SMALL_BUBBLES_IMG,
  smallGallaxy: SMALL_GALLAXY_IMG,
  logo: LOGO_IMG,
  people: PEOPLE_IMG,
};

export const ASTRO1_IMG_RATIO = 526 / 615; //used to maintain the original aspect ratio of the image.
export const PEOPLE_IMG_RATIO = 416 / 1209;
