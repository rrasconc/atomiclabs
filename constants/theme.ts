import { scale, verticalScale } from 'react-native-size-matters';

export const BORDER_RADIUS = 10;

export const FOOTER_HEIGHT = verticalScale(140);

export const SPACING = {
  extraSmall: scale(5),
  small: scale(10),
  medium: scale(15),
  large: scale(20),
};
export const FONT = {
  extraSmall: scale(13),
  small: scale(14),
  medium: scale(18),
  large: scale(32),
  extraLarge: scale(48),
};

export const COLORS = {
  background: '#072348',
  white: '#D6D6DD',
  orange: '#C24112',
  primary: '#00609C',
  black: '#000000',
  danger: '#CD1D29',
};
