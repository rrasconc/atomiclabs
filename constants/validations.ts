import { object, string } from 'yup';

export const signUpSchema = object({
  name: string().min(5, 'El nombre deberá tener mínimo 5 caracteres'),
  lastname: string().min(5, 'El apellido deberá tener mínimo 5 caracteres'),
  phoneNumber: string().min(9, 'Ingresa un número celular válido'),
});
