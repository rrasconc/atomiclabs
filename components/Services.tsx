import { LegacyRef, forwardRef } from 'react';
import { View, StyleSheet } from 'react-native';

import { HeaderText } from './Header.Text';
import { ServicesCarousel } from './Services.Carousel';
import { COLORS, SPACING } from '../constants/theme';

export const Services = forwardRef(function Services(_props, ref: LegacyRef<View>) {
  return (
    <View ref={ref} style={styles.container}>
      <View>
        <HeaderText>SOMOS EL BRAZO</HeaderText>
        <HeaderText>
          DERECHO<HeaderText style={styles.highlightText}> DE LA</HeaderText>
        </HeaderText>
        <HeaderText style={styles.highlightText}>TECNOLOG√çA</HeaderText>
      </View>

      <ServicesCarousel />
    </View>
  );
});

const styles = StyleSheet.create({
  container: {
    gap: SPACING.large * 2,
    marginVertical: SPACING.large,
  },
  highlightText: {
    color: COLORS.orange,
  },
});
