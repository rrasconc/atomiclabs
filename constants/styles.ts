import { StyleSheet } from 'react-native';

import { COLORS } from './theme';

export const globalStyles = StyleSheet.create({
  highlightText: {
    color: COLORS.orange,
  },
  label: {
    fontWeight: '500',
  },
});
