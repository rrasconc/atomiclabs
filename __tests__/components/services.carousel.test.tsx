import { render } from '@testing-library/react-native';

import { ServicesCarousel } from '../../components/Services.Carousel';
import { SERVICE_GROUPS } from '../../constants/content';

describe('<ServicesCarousel />', () => {
  let component: any = null;
  beforeEach(() => {
    component = render(<ServicesCarousel data={SERVICE_GROUPS} />);
  });

  it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
