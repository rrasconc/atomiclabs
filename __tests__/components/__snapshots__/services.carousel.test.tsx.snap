// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServicesCarousel /> renders correctly 1`] = `
<View
  style={
    {
      "gap": 42.857142857142854,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "justifyContent": "center",
      }
    }
    data={
      [
        {
          "image": 1,
          "services": [
            "Estrategia Digital",
            "Big Data & Analysis",
            "Consultoría Tecnológica",
            "Reducción de costos TI",
          ],
          "title": "IMAGINA",
        },
        {
          "image": 1,
          "services": [
            "Innovación y creación tecnológica",
            "UI/UX",
            "Innovación",
          ],
          "title": "EXPLORA",
        },
        {
          "image": 1,
          "services": [
            "Desarrollo tecnológico y creación tecnológica",
            "Ciberseguridad",
            "Servicios de la Nube",
          ],
          "title": "CONQUISTA",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={[]}
    style={
      {
        "marginHorizontal": -32.14285714285714,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            [
              {
                "paddingHorizontal": 32.14285714285714,
              },
              {
                "width": 750,
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FA4D09",
                "borderRadius": 10,
                "flex": 1,
                "gap": 32.14285714285714,
                "height": 784.7058823529411,
                "paddingHorizontal": 42.857142857142854,
                "paddingVertical": 32.14285714285714,
              }
            }
          >
            <Image
              resizeMode="cover"
              source={1}
              style={
                {
                  "height": 385.7142857142857,
                  "width": 385.7142857142857,
                }
              }
            />
            <View
              style={
                {
                  "gap": 21.428571428571427,
                  "paddingHorizontal": 42.857142857142854,
                }
              }
            >
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 10,
                    "height": 5.885294117647058,
                    "width": 300,
                  }
                }
              />
              <Text
                adjustsFontSizeToFit={true}
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "fontSize": 68.57142857142857,
                      "fontWeight": "700",
                      "textAlign": "center",
                    },
                    {
                      "textShadowColor": "rgba(0, 0, 0, 0.6)",
                      "textShadowOffset": {
                        "height": 1,
                        "width": -1,
                      },
                      "textShadowRadius": 10,
                    },
                  ]
                }
              >
                IMAGINA
              </Text>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  Estrategia Digital
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  Big Data & Analysis
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  Consultoría Tecnológica
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  Reducción de costos TI
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            [
              {
                "paddingHorizontal": 32.14285714285714,
              },
              {
                "width": 750,
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FA4D09",
                "borderRadius": 10,
                "flex": 1,
                "gap": 32.14285714285714,
                "height": 784.7058823529411,
                "paddingHorizontal": 42.857142857142854,
                "paddingVertical": 32.14285714285714,
              }
            }
          >
            <Image
              resizeMode="cover"
              source={1}
              style={
                {
                  "height": 385.7142857142857,
                  "width": 385.7142857142857,
                }
              }
            />
            <View
              style={
                {
                  "gap": 21.428571428571427,
                  "paddingHorizontal": 42.857142857142854,
                }
              }
            >
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 10,
                    "height": 5.885294117647058,
                    "width": 300,
                  }
                }
              />
              <Text
                adjustsFontSizeToFit={true}
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "fontSize": 68.57142857142857,
                      "fontWeight": "700",
                      "textAlign": "center",
                    },
                    {
                      "textShadowColor": "rgba(0, 0, 0, 0.6)",
                      "textShadowOffset": {
                        "height": 1,
                        "width": -1,
                      },
                      "textShadowRadius": 10,
                    },
                  ]
                }
              >
                EXPLORA
              </Text>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  Innovación y creación tecnológica
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  UI/UX
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  Innovación
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            [
              {
                "paddingHorizontal": 32.14285714285714,
              },
              {
                "width": 750,
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FA4D09",
                "borderRadius": 10,
                "flex": 1,
                "gap": 32.14285714285714,
                "height": 784.7058823529411,
                "paddingHorizontal": 42.857142857142854,
                "paddingVertical": 32.14285714285714,
              }
            }
          >
            <Image
              resizeMode="cover"
              source={1}
              style={
                {
                  "height": 385.7142857142857,
                  "width": 385.7142857142857,
                }
              }
            />
            <View
              style={
                {
                  "gap": 21.428571428571427,
                  "paddingHorizontal": 42.857142857142854,
                }
              }
            >
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 10,
                    "height": 5.885294117647058,
                    "width": 300,
                  }
                }
              />
              <Text
                adjustsFontSizeToFit={true}
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "fontSize": 68.57142857142857,
                      "fontWeight": "700",
                      "textAlign": "center",
                    },
                    {
                      "textShadowColor": "rgba(0, 0, 0, 0.6)",
                      "textShadowOffset": {
                        "height": 1,
                        "width": -1,
                      },
                      "textShadowRadius": 10,
                    },
                  ]
                }
              >
                CONQUISTA
              </Text>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  Desarrollo tecnológico y creación tecnológica
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  Ciberseguridad
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "gap": 21.428571428571427,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 50,
                      "height": 17.142857142857142,
                      "width": 17.142857142857142,
                    }
                  }
                />
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 38.57142857142857,
                      },
                      undefined,
                    ]
                  }
                >
                  Servicios de la Nube
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 32.14285714285714,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 20,
            "height": 32.14285714285714,
            "width": 32.14285714285714,
          },
          {
            "backgroundColor": "#FA4D09",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 20,
            "height": 32.14285714285714,
            "width": 32.14285714285714,
          },
          false,
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 20,
            "height": 32.14285714285714,
            "width": 32.14285714285714,
          },
          false,
        ]
      }
    />
  </View>
</View>
`;
